cilium/bpf_metadata.cc

```diff
@@ -546,7 +584,7 @@ Network::FilterStatus Instance::onAccept(Network::ListenerFilterCallbacks& cb) {
       }
 
       auto options = std::make_shared<Network::Socket::Options>();
-      options->push_back(std::move(option));
+      options->push_back(std::move(cilium_policy_socket_option));
```

Comment:
note: only the `CiliumPolicySocketOption` is set in the filterstate `UpstreamSocketOptionsFilterState` - and therefore passed to the upstream sockets.

AFAIU this is ok - and we don't have to add all the other options. Reasoning for this is that this functionality has been added with https://github.com/cilium/proxy/pull/581 - and the only reason mentioned is that the policy information need to be potentially available for upstream filters.